### PR TITLE
Add forms to edit and create user

### PR DESCRIPTION
**Pull request by flpfar**
This PR adds:
- User migrations and model with validations
- User controller
- HTML for new and edit pages.
- Basic routes to add and edit users
- Appropriate readme file

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      redirect_to new_user_path
+    else
+      p @user.errors.full_messages
```
Comment (suretrust) [inline app/controllers/users_controller.rb:12]: Let's remove the `p`, you can use flash message or save the error in an instance variable that can be called in the views.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  def update
+    @user = User.find(params[:id])
+    updated_user = user_params
+
+    updated_user[:password] = @user[:password] if updated_user[:password].empty?
+
+    if @user.update(updated_user)
+      redirect_to edit_user_path(@user)
+    else
+      p @user.errors.full_messages
```
Comment (suretrust) [inline app/controllers/users_controller.rb:30]: Same thing here.

**Comment by flpfar** [comment]
Thank you for reviewing our code, @suretrust . The issues have been fixed.

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    updated_user = user_params
+
+    updated_user[:password] = @user[:password] if updated_user[:password].empty?
```
Comment (suretrust) [inline app/controllers/users_controller.rb:27]: We can refactor this as `updated_user[:password] ||= @user[:password] `.

**Comment by sergiomauz** [comment]
Done! Thank you 😃

**Pull request merged**